### PR TITLE
fix: patch for version 2.2

### PR DESCRIPTION
**Pull request by ltsNotMike**
## Ticket

https://github.com/VirtusLab/strapi-plugin-navigation/issues/257
https://github.com/VirtusLab/strapi-plugin-navigation/issues/258
https://github.com/VirtusLab/strapi-plugin-navigation/issues/259

## Summary

Fixes for version 2.2
- Drag and drop not working on firefox. Fixed by introducing a custom drag button component.
- Audience disappearing from navigation items after saving. Fixed audience parsing on request.
- Slug not showing in the configuration on the Settings Page. Fixed by adding 'UID' type to configuration options. 

## Test Plan

How are you testing the work you're submitting?

- Open the admin panel on the newest Firefox. It should be possible to reorder items by dragging them.  
- Add some audience fields to the element. Try collapsing the item and saving the navigation. The audience should not disappear from the navigation item. 
- Add an UID field to the content type and try configuring it to be a default title or default path on the Settings Page.

### CODE REVIEW

**Review by cyp3rius — APPROVED** [review]

File: admin/src/components/Item/ItemCardHeader/index.tsx [hunk 1]
```diff
@@ -0,0 +1,61 @@
...
+        {path}
+      </Typography>
+      <Icon as={icon} />
+    </Flex>
+    <Flex alignItems="center" style={{ zIndex: 2 }}>
+      {removed &&
+        (<ItemCardBadge
+          borderColor={`danger200`}
+          backgroundColor={`danger100`}
+          textColor={`danger600`}
```
Comment (CodeVoyager) [inline admin/src/components/Item/ItemCardHeader/index.tsx:46]:
```suggestion
          borderColor="danger200"
          backgroundColor="danger100"
          textColor="danger600"
```

File: admin/src/components/Item/ItemCardHeader/index.tsx [hunk 2]
```diff
@@ -0,0 +1,61 @@
...
+      <DragButton ref={dragRef} />
+      <Typography variant="omega" fontWeight="bold">
+        {title}
+      </Typography>
+      <Typography variant="omega" fontWeight="bold" textColor='neutral500'>
+        {path}
+      </Typography>
+      <Icon as={icon} />
+    </Flex>
+    <Flex alignItems="center" style={{ zIndex: 2 }}>
```
Comment (CodeVoyager) [inline admin/src/components/Item/ItemCardHeader/index.tsx:41]:
```suggestion
    <Flex alignItems="center" style={wrapperStyles}>
```

or similar

File: admin/src/components/Item/ItemCardHeader/index.tsx [hunk 3]
```diff
@@ -0,0 +1,61 @@
...
+        >
+          {getMessage("components.navigationItem.badge.removed")}
+        </ItemCardBadge>)
+      }
+
+      <IconButton disabled={removed} onClick={onItemEdit} label="Edit" icon={<Pencil />} />
+      {removed ?
+        <IconButton onClick={onItemRestore} label="Restore" icon={<Refresh />} /> :
+        <IconButton onClick={onItemRemove} label="Remove" icon={<Trash />} />
+      }
```
Comment (CodeVoyager) [inline admin/src/components/Item/ItemCardHeader/index.tsx:56]: Consider moving icons to `icons.tsx`

**Review by CodeVoyager — APPROVED** [review]

File: admin/src/components/Item/ItemCardHeader/icons.tsx [hunk 4]
```diff
@@ -0,0 +1,7 @@
+import React from "react";
+// @ts-ignore
+import { Pencil, Trash, Refresh } from '@strapi/icons';
+
+export const pencilIcon = <Pencil />;
+export const refreshIcon = <Refresh />;
+export const trashIcon = <Trash />;
```
Comment (kamilszewczyk0) [inline admin/src/components/Item/ItemCardHeader/icons.tsx:7]: More question to discuss - are we adding "Icon" at the end of icon component name or we leave it without it? Cause we should set one standard for that

File: admin/src/components/Item/ItemCardHeader/index.tsx [hunk 5]
```diff
@@ -0,0 +1,63 @@
...
+interface IProps {
+  title: string,
+  path: string,
+  icon: ToBeFixed,
+  removed: boolean,
+  onItemRemove: VoidEffect,
+  onItemEdit: VoidEffect,
+  onItemRestore: VoidEffect,
+  dragRef: React.MutableRefObject<HTMLHeadingElement>,
+}
```
Comment (kamilszewczyk0) [inline admin/src/components/Item/ItemCardHeader/index.tsx:27]: move to separate file with types

**Review by kamilszewczyk0 — APPROVED** [review]

File: admin/src/components/Item/ItemCardHeader/icons.tsx (hunk 4 above)
Comment (ltsNotMike) [reply admin/src/components/Item/ItemCardHeader/icons.tsx:7]: I think it is better with the "Icon" suffix. It makes it less possible for a variable name collision to occur.

File: admin/src/components/Item/ItemCardHeader/index.tsx (hunk 5 above)
Comment (ltsNotMike) [reply admin/src/components/Item/ItemCardHeader/index.tsx:27]: I think it can be omitted here since there is only one type in this file.

**Pull request merged**